File: LeetCode/Easy/1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        t_0 = 0
        t_1 = 1
        t_2 = 1
        if n == 0: return t_0
        if n == 1: return t_1
        if n == 2: return t_2

        for _ in range(n - 2):
            t_0, t_1, t_2 = t_1, t_2, t_0 + t_1 + t_2
        
        return t_2

File: LeetCode/Easy/1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cs
public class Solution {
    public int Tribonacci(int n) {
        var t0 = 0;
        var t1 = 1;
        var t2 = 1;

        if (n == 0) return t0;
        if (n == 1) return t1;
        if (n == 2) return t2;

        for (var i = 0; i < n - 2; i++) (t0, t1, t2) = (t1, t2, t0 + t1 + t2);

        return t2;
    }
}